crates/compiler/gen_llvm/src/llvm/refcounting.rs

```diff
@@ -986,7 +986,9 @@ pub fn build_header_help<'a, 'ctx, 'env>(
         env.module,
         fn_name,
         FunctionSpec::known_fastcc(fn_type),
-        Linkage::Private,
+        // this should be `Linkage::Private`, but that will remove all of the code for the inc/dec
+        // functions on windows. LLVM just does not emit the assembly for them. Investigate why this is
+        Linkage::External,
```

Comment:
this is truly strange: with our standard private linking, LLVM just does not emit the assembly. But it is not DCO because functions that only dec calls, do still have assembly. 

This is a quick fix, but means that inc/dec are never DCO'd if they are unused.